docs/Development.md

```diff
@@ -0,0 +1,104 @@
+
+Uses of CMake on non-C/C++ codebases are usually documented along with corresponding codebases. The general rule is that after running CMake configure and CMake build the [global runner](./GlobalRunner.md) should work correctly, and running CTest should execute all existing unit tests.
+
+## Adding New Components
+
+### Testing Modules
+
+If you want to add a testing module for one of the existing benchmark runners, then, please, refer to the corresponding documentation:
+
+- [C++](cpp/Modules.md)
+- [.NET Core](dotnet/Modules.md)
+- [Julia](julia/Modules.md)
+- [Python](python/Modules.md)
+
+Otherwise, you'll need to implement a benchmark runner for your module first.
+
+### Benchmark Runners
+
+The description of what benchmark runners are and what they should do can be found [here](./Architecture.md#benchmark-runners). Probably, the easiest way to implement one is to translate one of the existing benchmark runners into the language you need. The part that dynamically loads plugins will obviously need adaptation to your target platform, though.
+
+When looking for where to place your new code, follow the patterns established for other platforms (e.g. sources in `src/<platform>/**`, tests in `test/<platform>/**`, etc.).
+
+Write `CMakeLists.txt` files, so that all your dependencies are fetched when necessary, your compiled code is built when CMake build is invoked, and your unit tests is executed by CTest with all other unit tests. Writing documentation similar to that for the existing runners would also be very appreciated.
+
+Make the [global runner](./GlobalRunner.md) be able to work with your benchmark runner. See the [corresponding section](./GlobalRunner.md#interfacing-with-benchmark-runners) of its documentation for details on how to do that.
+
+### Objective Functions
+
+To add a new objective functions, start by adding its support to benchmark runners. For details, see the corresponding documentation for each runner:
+- [C++](cpp/Runner.md#adding-new-objective-types)
+- [.NET Core](dotnet/Runner.md#adding-new-objective-types)
+- [Julia](julia/Runner.md#adding-new-objective-types)
+- [Python](python/Runner.md#adding-new-objective-types)
+
+After that add the support for your new objective functions to all testing modules you want to benchmark with it.
+
+Put inputs for your new objective function into the corresponding subfolder of `data` folder.
+
+Finally, add the it to `run-all.ps1`:
+1. Add the name of your objective to the `ObjectiveType` enumeration.
+1. Create a new method on the `[Tool]` class (format should be `[Tool]::test<your objective name>()`)
```

Comment:
Seems like a misprint: `a new method on the [Tool] class` instead of `a new method of the [Tool] class`

Reply:
Fixed.